src/uk/gov/hmcts/contino/TeamNames.groovy

```diff
@@ -2,77 +2,59 @@ package uk.gov.hmcts.contino
 
 class TeamNames {
 
+  def steps
+  static final String GITHUB_CREDENTIAL = 'jenkins-github-hmcts-api-token'
   static final String DEFAULT_TEAM_NAME = 'pleaseTagMe'
+  static final String NAMESPACE_KEY = "namespace"
+  static final String TEAM_KEY = "team"
+  static def teamNamesMap
 
-  def teamNamesMap = [
+  TeamNames (steps){
+    this.steps = steps
+  }
+
+  def getTeamNamesMap() {
+    if (teamNamesMap ==null ){
+      def response = steps.httpRequest(
+        consoleLogResponseBody: true,
+        authentication: "${GITHUB_CREDENTIAL}",
+        timeout: 10,
+        url: "https://raw.githubusercontent.com/hmcts/cnp-jenkins-config/master/team-config.yml",
```

Comment:
this PR looks great, only question would be is that the most appropriate place?
i.e. should we use a generic config repo, something like:
azure-team-config ? @njrich28

Reply:
Decided to continue in there for now.